packages/eslint-plugin/tests/rules/component-selector.test.ts

```diff
@@ -341,5 +373,33 @@ ruleTester.run(RULE_NAME, rule, {
       ],
       data: { type: 'attribute' },
     }),
+    convertAnnotatedSourceToFailureCase({
+      description: `it should fail if a ShadowDom selector is not kebab-case`,
+      annotatedSource: `
+      @Component({
+        encapsulation: ViewEncapsulation.ShadowDom,
+        selector: 'appFooBar'
+                  ~~~~~~~~~~~
+      })
+      class Test {}
+      `,
+      messageId: messageIdStyleFailure,
```

Comment:
I can see users being easily confused and maybe even creating issues because of this report message, because let's say that one set `style` to `camelCase`, but the report message says:

> The selector must be kebab-case (link)

That is, something that is not in accordance with the established configuration. I don't know... how about creating a specific message for ShadowDom-encapsulated components that says something like

> The selector of a ShadowDom-encapsulated component should be `kebab-case` (link).

cc @JamesHenry

Reply:
@rafaelss95 Good point, I suppose we could define an additional message-id `shadowDomStyleFailure` to report shadowdom-encapsulated component selector style errors, and report this error by calling `context.report()` directly. Does it make sense?
cc: @JamesHenry

Reply:
Yes I agree that a separate clear error message would be better 👍